src/domain/transaction/evm/price_calculator.rs

```diff
@@ -200,59 +180,21 @@ where
         tx_data: &EvmTransactionData,
         relayer: &RelayerRepoModel,
     ) -> Result<PriceParams, TransactionError> {
-        let price_params = self
+        let mut final_params = self
```

Comment:
Just to verify flow in case zkevm is used and speed params are used(so we are fetching from rpc)

At first sight it seems we would use standard json rpc methods to calculate price in `fetch_price_params_based_on_tx_type` and later we use zkevm handler to overwrite that data. 

Seems like we have unnecessary step and calculating twice. Could you confirm if that is the flow.